### PR TITLE
[PHPStan] level 2 fixes

### PR DESCRIPTION
**Pull request by weisswurstkanone**

### CODE REVIEW

File: models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php [hunk 1]
```diff
@@ -69,9 +74,20 @@ public function setClasses($classes)
      */
     public function getLazyLoading()
     {
-        return true;
+        return $this->lazyLoading;
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: why?

Comment (brusch) [reply models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: Doesn't make sense, needs to be reverted 😊

File: models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php [hunk 2]
```diff
@@ -72,6 +77,17 @@ public function getLazyLoading()
         return true;
     }
 
+    /**
+     * @param bool $lazyLoading
+     *
+     * @return $this
+     */
+    public function setLazyLoading($lazyLoading)
```
Comment (dvesh3) [inline models/DataObject/ClassDefinition/Data/Relations/AbstractRelations.php]: what about other changes in this file?

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php [hunk 3]
```diff
@@ -205,6 +205,7 @@ public function getDataForQueryResource($data, $object = null, $params = [])
         $d = [];
 
         if (is_array($data) && count($data) > 0) {
+            /** @var DataObject\Data\ElementMetadata $metaObject */
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: Than we can set the $data param to `DataObject\Data\ElementMetadata[]|null`, right?

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: isn't this already the case?

Comment (blankse) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: No it's `Element\ElementInterface[]|null`. This var comment should no longer be necessary after that.

Comment (weisswurstkanone) [reply models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]: Ah, now I got you! thx

File: models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php [hunk 4]
```diff
@@ -186,7 +186,7 @@ public function loadData($data, $object = null, $params = [])
     }
 
     /**
-     * @param Element\ElementInterface[]|null $data
+     * @param Element\ElementMetadata[]|null $data
```
Comment (blankse) [inline models/DataObject/ClassDefinition/Data/AdvancedManyToManyRelation.php]:
```suggestion
     * @param DataObject\Data\ElementMetadata[]|null $data
```

**Pull request merged**